crates/rocksdb/src/lib.rs

```diff
@@ -0,0 +1,242 @@
+        &self.db
+    }
+}
+
+impl RocksDb {
+    pub(crate) fn new<T>(spec: DbSpec<T>, db: Arc<T>) -> Self
+    where
+        T: RocksAccess + Send + Sync + 'static,
+    {
+        Self {
+            owner: spec.owner,
+            name: spec.name,
+            db,
+            db_options: spec.db_options,
+            column_families: spec.column_families.into_iter().collect(),
+        }
+    }
+
+    /// Returns the raw rocksdb handle, this should only be used for server operations that
+    /// require direct access to rocksdb.
+    pub fn inner(&self) -> &Arc<dyn RocksAccess + Send + Sync + 'static> {
+        &self.db
+    }
+
+    pub fn cfs(&self) -> Vec<&CfName> {
+        self.column_families.keys().collect()
+    }
+
+    pub fn get_histogram_data(&self, histogram: Histogram) -> HistogramData {
+        self.db_options.get_histogram_data(histogram)
+    }
+
+    pub fn get_ticker_count(&self, ticker: Ticker) -> u64 {
+        self.db_options.get_ticker_count(ticker)
+    }
+
+    pub fn get_statistics_str(&self) -> Option<String> {
+        self.db_options.get_statistics()
+    }
+}
+
+pub(crate) fn amend_db_options(db_options: &mut rocksdb::Options, opts: &RocksDbOptions) {
+    db_options.create_if_missing(true);
+    db_options.create_missing_column_families(true);
+    db_options.set_atomic_flush(true);
+
+    if opts.rocksdb_num_threads() > 0 {
+        db_options.increase_parallelism(opts.rocksdb_num_threads() as i32);
+        db_options.set_max_background_jobs(opts.rocksdb_num_threads() as i32);
+    }
+
+    if !opts.rocksdb_disable_statistics() {
+        db_options.enable_statistics();
+        // Reasonable default, but we might expose this as a config in the future.
+        db_options.set_statistics_level(rocksdb::statistics::StatsLevel::ExceptDetailedTimers);
+    }
+
+    db_options.set_atomic_flush(true);
+    //
+    // Disable WAL archiving.
+    // the following two options has to be both 0 to disable WAL log archive.
+    //
+    db_options.set_wal_size_limit_mb(0);
+    db_options.set_wal_ttl_seconds(0);
+    //
+    if !opts.rocksdb_disable_wal() {
+        // Disable automatic WAL flushing.
+        // We will call flush manually, when we commit a storage transaction.
+        //
+        db_options.set_manual_wal_flush(opts.rocksdb_batch_wal_flushes());
+        // Once the WAL logs exceed this size, rocksdb start will start flush memtables to disk
+        // We set this value to 10GB by default, to make sure that we don't flush
+        // memtables prematurely.
+        db_options.set_max_total_wal_size(opts.rocksdb_max_total_wal_size());
+    }
+    //
+    // Let rocksdb decide for level sizes.
+    //
+    db_options.set_level_compaction_dynamic_level_bytes(true);
+    db_options.set_compaction_readahead_size(1 << 21);
+    //
+    // We use Tokio's background threads to access rocksdb, and
+    // at most we have 512 threads.
+    //
+    db_options.set_table_cache_num_shard_bits(9);
```

Comment:
I think this is no longer the case that we are using Tokio's background threads.

Reply:
I'll update the comment. I think we still often _read_ from rocksdb in tokio threads (although not background threads)